src/Updater.php

```diff
@@ -327,6 +327,13 @@ protected function newVersionAvailable()
 
     protected function backupPhar()
     {
+        $tempDirectory = $this->getTempDirectory();
```

Comment:
> Concerning #1, I don't think that will work, because getTempDirectory() method is called in the update check process. That would cause the update check to fail in the same way that required the change (Issue #20 ).

I've tracked back every call of `\Humbug\SelfUpdate\Updater::getTempDirectory` and each of them is prefixed by exception throwing when path isn't writable in one way or another. So moving exception throwing into that method would reduce duplication and preserve logic introduced in this PR.



> Concerning #2, here in line 330 I am using the getTempDirectory() method as requested and assigning it to a local variable in the method's scope. I don't see a place where I am using the tempDirectory property directly, but please clarify if you see a line where that is done.

I've double checked again and the place is gone. Probably something in the code was changed or I've written my comment theoretically (without looking at the code).